youtube_dl/extractor/facebook.py

```diff
@@ -509,3 +512,116 @@ def _real_extract(self, url):
         return self.url_result(
             compat_urllib_parse_unquote(self._match_id(url)),
             FacebookIE.ie_key())
+
+
+class FacebookUserIE(InfoExtractor):
+    _VALID_URL = r'(?P<url>https?://(?:[^/]+\.)?facebook\.com/(?:pg/)?(?P<id>[^/?#&]+))/videos(?!/[\w\.])'
+    IE_NAME = "facebook:user"
+
+    _TESTS = [{
+        # page
+        'url': 'https://www.facebook.com/Coca-Cola/videos/?ref=page_internal',
+        'info_dict': {
+            'id': 'Coca-Cola'
+        },
+        'playlist_mincount': 90,
+    }, {
+        # profile (requires login)
+        'url': 'https://www.facebook.com/zuck/videos',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        user_id = mobj.group('id')
+
+        page = self._download_webpage(
+            url, user_id, 'Downloading user webpage')
+        fb_url = self._html_search_meta(
+            'al:android:url', page, default=None)
+        fb_url_mobj = re.match(r'fb://(?P<type>page|profile)/(?P<id>\d+)', fb_url)
+        if not fb_url_mobj:
+            raise ExtractorError('Could not extract page ID', expected=False)
+        page_id = fb_url_mobj.group('id')
+        fb_dtsg_ag = self._search_regex(
+            r'"async_get_token":"([\w\-:]+)"',
```

Comment:
Relax.

Reply:
How should I relax the regex? The tokens only contain those characters

Reply:
@dstftw How should I relax the regex? The tokens only contain those characters